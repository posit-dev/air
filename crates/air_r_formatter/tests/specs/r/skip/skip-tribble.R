#' [format]
#' skip = ["tribble"]

# Should skip formatting even without a skip comment
tribble(
  ~x, ~y,
   1,  2,
   3,  4
)

# "Sees through" namespaces and looks at the function name
tibble::tribble(
  ~x, ~y,
   1,  2,
   3,  4
)

# Purposefully not smart enough to see through this
(tribble)(
  ~x, ~y,
   1,  2,
   3,  4
)

# Custom function would have to be added to the skip list to get skipped
tzibble <- function(...) {
    tribble(...)
}
tzibble(
  ~x, ~y,
   1,  2,
   3,  4
)
