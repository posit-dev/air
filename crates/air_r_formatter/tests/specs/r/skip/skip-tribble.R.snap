---
source: crates/air_formatter_test/src/snapshot_builder.rs
info: r/skip/skip-tribble.R
---
# Input

```R
#' [format]
#' skip = ["tribble"]

# Should skip formatting even without a skip comment
tribble(
  ~x, ~y,
   1,  2,
   3,  4
)

# "Sees through" namespaces and looks at the function name
tibble::tribble(
  ~x, ~y,
   1,  2,
   3,  4
)

# Purposefully not smart enough to see through this
(tribble)(
  ~x, ~y,
   1,  2,
   3,  4
)

# Custom function would have to be added to the skip list to get skipped
tzibble <- function(...) {
    tribble(...)
}
tzibble(
  ~x, ~y,
   1,  2,
   3,  4
)

```


=============================

# Outputs

## Output 1

-----
Indent style: Space
Indent width: 2
Line ending: LF
Line width: 80
Persistent line breaks: Respect
Skip: tribble
-----

```R
#' [format]
#' skip = ["tribble"]

# Should skip formatting even without a skip comment
tribble(
  ~x, ~y,
   1,  2,
   3,  4
)

# "Sees through" namespaces and looks at the function name
tibble::tribble(
  ~x, ~y,
   1,  2,
   3,  4
)

# Purposefully not smart enough to see through this
(tribble)(
  ~x,
  ~y,
  1,
  2,
  3,
  4
)

# Custom function would have to be added to the skip list to get skipped
tzibble <- function(...) {
  tribble(...)
}
tzibble(
  ~x,
  ~y,
  1,
  2,
  3,
  4
)
```
